hubclient/errors.go

```diff
@@ -98,3 +98,10 @@ func HubClientErrorf(format string, args ...interface{}) error {
 	err := &HubClientError{newErr, 0, HubResponseError{}}
 	return err
 }
+
+func HubClientStatusCodeErrorf(statusCode int, format string, args ...interface{}) error {
```

Comment:
The function above does exactly same with status 0, you can call this one from there.